index.d.ts

```diff
@@ -0,0 +1,49 @@
+import { 
+    Event, 
+    EventTarget, 
+    AddEventListenerOptions, 
+    EventListenerOptions,
+    EventTargetListener
+ } from "event-target-shim";
+
+export as namespace AbortControllerShim;
+
+/**
+ * Temporary event type to keep typescript happy.
+ */
+interface ProgressEvent extends Event {
+    readonly lengthComputable: any;
+    readonly loaded: any;
+    readonly total: any;
+}
+
+export type AbortSignalListener = (this: AbortSignal, ev: ProgressEvent) => any;
```

Comment:
Does this definition cause a type error in the future? If yes, I want to avoid it. It's the reason that I suggested `any`.

Reply:
It will have to be changed to `Event` once TS3.3 comes out.

Reply:
Yes. But I want to avoid type errors without the change.

Reply:
Hmm, fine by me. I'll go ahead and open issue here and in event-shim to change back `any` types once TS3.3 comes out.